packages/astro/src/core/build/plugins/plugin-css.ts

```diff
@@ -251,9 +250,16 @@ function rollupPluginAstroBuildCSS(options: PluginOptions): VitePlugin[] {
 						// return early if the stylesheet needing propagation has already been included
 						if (pageData.styles.some((s) => s.sheet === sheet)) return;
 
-						const propagatedStyles =
-							internals.propagatedStylesMap.get(pageInfoId) ??
-							internals.propagatedStylesMap.set(pageInfoId, new Set()).get(pageInfoId)!;
+						let propagatedStyles: Set<StylesheetAsset>;
+						if (isContentCollectionCache) {
+							propagatedStyles =
+								internals.propagatedStylesMap.get(pageInfoId) ??
+								internals.propagatedStylesMap.set(pageInfoId, new Set()).get(pageInfoId)!;
+						} else {
+							propagatedStyles =
+								pageData.propagatedStyles.get(pageInfoId) ??
+								pageData.propagatedStyles.set(pageInfoId, new Set()).get(pageInfoId)!;
+						}
```

Comment:
So `internals` is the new place to store propagated assets, but it was read from even without the opt-in, when it should've been the current place `pageData` instead. Is that what happened?

Reply:
It was a few things! But that's the main one. Reading from `internals` should have only been behind the flag as the current behavior relies on `pageData.propagatedStyles`. I missed this when refactoring to put the feature back behind a flag and mistakenly assumed we had test coverage on this feature.